framework/auth/core.py

```diff
@@ -843,6 +849,35 @@ def _get_unconfirmed_email_for_token(self, token):
 
         return verification['email']
 
+    def clean_email_verifications(self):
+        email_verifications = deepcopy(self.email_verifications)
+        for token in self.email_verifications:
+            try:
+                self.get_unconfirmed_email_for_token(token)
+            except (KeyError, ExpiredTokenError):
+                email_verifications.pop(token)
+        self.email_verifications = email_verifications
+
+    def unconfirmed_email_get(self):
```

Comment:
@sloira, where would be a good place to put clean_email_verifications?  auth_login?